### ISSUE TITLE
manual/scalaGuide/main/http/ScalaSessionFlash.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#1011]
- https://github.com/playframework-ja/translation-project/tree/2.3.x/manual/scalaGuide/main/http/ScalaSessionFlash.md
- diffs between [2.2.0/manual/scalaGuide/main/http/ScalaSessionFlash.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/scalaGuide/main/http/ScalaSessionFlash.md) and [2.3.x/manual/scalaGuide/main/http/ScalaSessionFlash.md](https://github.com/playframework-ja/translation-project/tree/2.3.x/manual/scalaGuide/main/http/ScalaSessionFlash.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/scalaGuide/main/http/ScalaSessionFlash.md   2013-09-19 10:11:22.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/scalaGuide/main/http/ScalaSessionFlash.md  2014-07-28 12:26:38.000000000 +0900
@@ -1,29 +1,20 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Session and Flash scopes

 ## How it is different in Play

 If you have to keep data across multiple HTTP requests, you can save them in the Session or Flash scopes. Data stored in the Session are available during the whole user Session, and data stored in the Flash scope are available to the next request only.

-It’s important to understand that Session and Flash data are not stored by the server but are added to each subsequent HTTP request, using the cookie mechanism. This means that the data size is very limited (up to 4 KB) and that you can only store string values.
+It’s important to understand that Session and Flash data are not stored by the server but are added to each subsequent HTTP request, using the cookie mechanism. This means that the data size is very limited (up to 4 KB) and that you can only store string values. The default name for the cookie is `PLAY_SESSION`. This can be changed by configuring the key `session.cookieName` in application.conf.
+
+> If the name of the cookie is changed, the earlier cookie can be discarded using the same methods mentioned in [[Setting and discarding cookies|ScalaResults]].

 Of course, cookie values are signed with a secret key so the client can’t modify the cookie data (or it will be invalidated).

-The Play Session is not intended to be used as a cache. If you need to cache some data related to a specific Session, you can use the Play built-in cache mechanism and use store a unique ID in the user Session to keep them related to a specific user.
+The Play Session is not intended to be used as a cache. If you need to cache some data related to a specific Session, you can use the Play built-in cache mechanism and store a unique ID in the user Session to keep them related to a specific user.

 > There is no technical timeout for the Session. It expires when the user closes the web browser. If you need a functional timeout for a specific application, just store a timestamp into the user Session and use it however your application needs (e.g. for a maximum session duration, maximum inactivity duration, etc.).

-## Reading a Session value
-
-You can retrieve the incoming Session from the HTTP request:
-
-@[index-retrieve-incoming-session](code/ScalaSessionFlash.scala)
-
-
-Alternatively you can retrieve the Session implicitly from a request:
-
-@[index-retrieve-incoming-session-implicitly](code/ScalaSessionFlash.scala)
-
-
 ## Storing data in the Session

 As the Session is just a Cookie, it is also just an HTTP header. You can manipulate the session data the same way you manipulate other results properties:
@@ -40,6 +31,12 @@

 @[remove-session](code/ScalaSessionFlash.scala)

+## Reading a Session value
+
+You can retrieve the incoming Session from the HTTP request:
+
+@[index-retrieve-incoming-session](code/ScalaSessionFlash.scala)
+
 ## Discarding the whole session

 There is special operation that discards the whole session:
@@ -73,4 +70,4 @@

 @[find-noflash](code/ScalaSessionFlash.scala)

-> **Next:** [[Body parsers | ScalaBodyParsers]]
\ No newline at end of file
+> **Next:** [[Body parsers | ScalaBodyParsers]]

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged at #1083
